src/View/Cell.php

```diff
@@ -176,6 +176,17 @@ public function render($template = null)
         }
 
         $render = function () use ($template) {
+            try {
+                $reflect = new ReflectionMethod($this, $this->action);
+                $reflect->invokeArgs($this, $this->args);
```

Comment:
From my understanding, there's nothing in the anonymous ``render`` function that stops the action from being run in [its current form](https://github.com/cakephp/cakephp/blob/master/src/View/Cell.php#L178-L214) if it is called.
But whether or not we resolve the template name before or after the cell action method is run does not change anything in the current form of the method.

Basically, the anonymous render function is run (and therefore the cell action is run) only if :
- there is no cache configured (we then [hit this - link from my PR](https://github.com/cakephp/cakephp/blob/issue-8163/src/View/Cell.php#L219))
- cache is configured but the cache does not exists (and it is then [executed by the ``Cache::remember()`` method here](https://github.com/cakephp/cakephp/blob/master/src/Cache/Cache.php#L560))

If the cache is configured and data exists, then ``Cache::remember()`` will render the cell without calling the action since it will not call the anonymous function.

If you had not refactored the cache to use ``remember()`` (meaning the render method would have been [in that form](https://github.com/cakephp/cakephp/blob/515c42e96f8006abc9257f93833c39ae40b7a241/src/View/Cell.php) (this is your first implementation of the cache cell), I would have agreed and not made the change.

So this should be good, unless there is something I'm missing (which might be the case).
I can try to add a test tonight to make sure of this if needed.